core/src/main/scala/org/apache/spark/deploy/security/HadoopFSDelegationTokenProvider.scala

```diff
@@ -121,6 +121,15 @@ private[deploy] class HadoopFSDelegationTokenProvider
     creds
   }
 
+  private def cancelDelegationTokens(hadoopConf: Configuration,
+                                     creds: Credentials) : Unit = {
+
+    creds.getAllTokens
+      .asScala
+      .filter(_.decodeIdentifier().isInstanceOf[AbstractDelegationTokenIdentifier])
+      .foreach { token => token.cancel(hadoopConf) }
+  }
```

Comment:
Maybe this? A little more Scala-idiomatic to use a pattern match instead of `isInstanceOf` 
```suggestion
    creds.getAllTokens.asScala.foreach {
      case id: AbstractDelegationTokenIdentifier => id.cancel(hadoopConf)
      case _ => // ignore
    }
```

Reply:
Oops this won't compile.
Let me try this instead:

creds.getAllTokens.asScala.foreach {
      case id if id.decodeIdentifier().isInstanceOf[AbstractDelegationTokenIdentifier] =>
        id.cancel(hadoopConf)
      case _ => // ignore
    }

Reply:
What's the error? that looks fine. The proposed change really isn't simpler than the original

Reply:
Hi Sean, the github action in my repo aborted with
"[error] /home/runner/work/spark/spark/core/src/main/scala/org/apache/spark/deploy/security/HadoopFSDelegationTokenProvider.scala:150:16: fruitless type test: a value of type org.apache.hadoop.security.token.Token[?0] cannot also be a org.apache.hadoop.security.token.delegation.AbstractDelegationTokenIdentifier
[error]       case id: AbstractDelegationTokenIdentifier => id.cancel(hadoopConf)
[error]                ^
[error] one error found"

https://github.com/jojochuang/spark/runs/4232847626?check_suite_focus=true

Reply:
Oh right. I'd go back to the original